app/functional_tests_fixtures/__init__.py

```diff
@@ -0,0 +1,460 @@
+        Permission(service_id=service.id, user_id=user.id, permission=SEND_TEXTS),
+        Permission(service_id=service.id, user_id=user.id, permission=SEND_EMAILS),
+    ]
+
+    permission_dao.set_user_service_permission(user, service, permission_list, _commit=True, replace=True)
+
+
+def _create_api_key(name, service_id, user_id, key_type="normal"):
+
+    api_keys = get_model_api_keys(service_id=service_id)
+    for key in api_keys:
+        if key.name == name:
+            return key
+
+    request = {"created_by": user_id, "key_type": key_type, "name": name}
+
+    fetched_service = dao_fetch_service_by_id(service_id=service_id)
+    valid_api_key = api_key_schema.load(request)
+    valid_api_key.service = fetched_service
+    save_model_api_key(valid_api_key)
+
+    return valid_api_key
+
+
+def _create_inbound_numbers(service_id, user_id, number="07700900500", provider="mmg"):
+
+    inbound_number = dao_get_inbound_number_for_service(service_id=service_id)
+
+    if inbound_number is not None:
+        return inbound_number.id
+
+    inbound_number = InboundNumber()
+    inbound_number.number = number
+    inbound_number.provider = provider
+    inbound_number.service_id = service_id
+    inbound_number.active = True
+    inbound_number.created_by = user_id
+
+    dao_set_inbound_number_to_service(service_id, inbound_number)
+
+    return inbound_number.id
+
+
+def _create_email_template(service, user_id):
+    name = "Functional Tests - CSV Email Template with Build ID"
+
+    templates = dao_get_all_templates_for_service(service_id=service.id)
+
+    for template in templates:
+        if template.name == name:
+            return template.id
+
+    data = {
+        "name": name,
+        "template_type": "email",
+        "content": "The quick brown fox jumped over the lazy dog. Build id: ((build_id)).",
+        "subject": "Functional Tests - CSV Email",
+        "created_by": user_id,
+    }
+
+    new_template = template_schema.load(data)
+
+    if not new_template.postage and new_template.template_type == LETTER_TYPE:
+        new_template.postage = SECOND_CLASS
+
+    new_template.service = service
+
+    dao_create_template(new_template)
+
+    return new_template.id
+
+
+def _create_sms_template(service, user_id):
+    name = "Functional Tests - CSV SMS Template with Build ID"
+
+    templates = dao_get_all_templates_for_service(service_id=service.id)
+
+    for template in templates:
+        if template.name == name:
+            return template.id
+
+    data = {
+        "name": "Functional Tests - CSV SMS Template with Build ID",
+        "template_type": "sms",
+        "content": "The quick brown fox jumped over the lazy dog. Build id: ((build_id)).",
+        "created_by": user_id,
+    }
+
+    new_template = template_schema.load(data)
+
+    if not new_template.postage and new_template.template_type == LETTER_TYPE:
+        new_template.postage = SECOND_CLASS
```

Comment:
we shouldn't need this guard in `_create_sms_template` right? is this a copy+paste error?

Reply:
Yeah copy+paste issue. Fixed.